redaxo/src/addons/cronjob/install.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+rex_sql::factory()->setQuery('
+    CREATE TABLE IF NOT EXISTS `'.rex::getTable('cronjob').'` (
+        id int(10) unsigned NOT NULL AUTO_INCREMENT,
+        PRIMARY KEY (`id`)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+');
+
+$table = rex_sql_table::get(rex::getTable('cronjob'));
+$table
+    ->ensureColumn(new rex_sql_column('id', 'int(10) unsigned', false, null, 'auto_increment'))
+    ->ensureColumn(new rex_sql_column('name', 'varchar(255)', true))
+    ->ensureColumn(new rex_sql_column('description', 'varchar(255)', true))
+    ->ensureColumn(new rex_sql_column('type', 'varchar(255)', true))
+    ->ensureColumn(new rex_sql_column('parameters', 'text', true))
+    ->ensureColumn(new rex_sql_column('interval', 'text'))
+    ->ensureColumn(new rex_sql_column('nexttime', 'datetime', true))
+    ->ensureColumn(new rex_sql_column('environment', 'varchar(255)'))
+    ->ensureColumn(new rex_sql_column('execution_moment', 'tinyint(1)'))
+    ->ensureColumn(new rex_sql_column('execution_start', 'datetime'))
+    ->ensureColumn(new rex_sql_column('status', 'tinyint(1)'))
+    ->ensureColumn(new rex_sql_column('createdate', 'datetime'))
+    ->ensureColumn(new rex_sql_column('createuser', 'varchar(255)'))
+    ->ensureColumn(new rex_sql_column('updatedate', 'datetime'))
+    ->ensureColumn(new rex_sql_column('updateuser', 'varchar(255)'))
+;
+if (method_exists($table, 'ensure')) {
+    $table->setPrimaryKey('id')->ensure();
```

Comment:
Deswegen habe ich es auch erst mal nur exemplarisch an einem Addon gemacht, um überhaupt erst mal zu klären, ob es das wert ist.
Können ja auch sagen, wir lassen all unsere Install-Routinen und update-Mechanismen erst mal so wie sie sind.

Wie im Eingangsposting beschrieben, haben es Dritt-Addons auch einfacher. Da würde ich empfehlen, den Weg erst zu gehen, wenn man auch die Redaxo-Mindestversion anhebt auf 5.4.
Und dann halt direkt diese Variante (noch mit den neuen Shortcuts): https://github.com/redaxo/redaxo/pull/1198#discussion_r109857392

Bei den Core-Addons haben wir es wie gesagt schwerer, und können das erst machen, wenn die Ausgangs-Redaxo-Version vor dem Update bereits mindestens 5.4 ist.

Noch ne Alternative: Die install.php nach neuer Variante schreiben, diese aber nicht in der `update.php` einbinden. Somit müssten wir dann auch zukünftig in der update.php alle Änderungen auch noch notieren.
Aber wäre durchaus auch ein vielleicht guter Weg.